mitm_service/mitm_service.py

```diff
@@ -0,0 +1,83 @@
+import binascii
+import socket
+import struct
+import fcntl
+
+from mitm_service.arp_poison.arp_poison import ARPPoisonService
+from mitm_service.tunneler.tunneler import L2Tunnel
+
+SIOCGIFHWADDR = 0x8927
+
+def get_interface_mac(interface_name):
+    """
+    Gets the mac string of a given interface
+    :param interface_name: the name of the interface who's mac to get
+    :return: the mac address of the interface as a string
+    """
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    info = fcntl.ioctl(s.fileno(), SIOCGIFHWADDR,  struct.pack('256s', bytes(interface_name, 'utf-8')[:15]))
+    return ':'.join('%02x' % b for b in info[18:24])
+
+class MITMService:
+    def __init__(self, interface, target_ip, gateway_ip, target_mac=None, gateway_mac=None):
+        self.interface = interface
+        self.target_ip = target_ip
+        self.gateway_ip = gateway_ip
+        self.my_mac = get_interface_mac(self.interface)
+
+        if target_mac is None:
+            self.target_mac = ARPPoisonService.get_mac(self.target_ip, 5)
+        else:
+            self.target_mac = target_mac
+        if gateway_mac is None:
+            self.gateway_mac = ARPPoisonService.get_mac(self.gateway_ip, 5)
+        else:
+            self.gateway_mac = gateway_mac
```

Comment:
self.target_mac = target_mac or type(self).get_mac() is much cleaner

Reply:
type(self) does not have a get_mac function, so it'd have to stay as ARPPoisonService.get_mac